pywayland/scanner/interface.py

```diff
@@ -129,47 +99,56 @@ def output(self, printer: Printer, module_imports: dict[str, str]) -> None:
                 self.description.output(printer)
                 printer('"""')
                 printer()
-
             # Class attributes
             printer(f'name = "{self.name}"')
             printer(f"version = {self.version}")
-
             # Enums
             for enum in self.enum:
                 printer()
                 enum.output(printer)
 
-        proxy_class_name = f"{self.class_name}Proxy"
-        resource_class_name = f"{self.class_name}Resource"
-        global_class_name = f"{self.class_name}Global"
-
+    def output_requests(self, printer: Printer) -> None:
+        """Generate the output only of the proxy class"""
         printer()
         printer()
+        proxy_class_name = f"{self.class_name}Proxy"
         printer(f"class {proxy_class_name}(Proxy[{self.class_name}]):")
         with printer.indented():
             printer(f"interface = {self.class_name}")
             for opcode, request in enumerate(self.request):
                 printer()
-                request.output(printer, opcode, self.class_name, module_imports)
+                request.output(printer, opcode, self.class_name)
+        printer()
+        printer()
+        printer(f"{self.class_name}.proxy_class = {proxy_class_name}")
```

Comment:
I think it might be a bit cleaner to move all of these interface setup functions to the bottom of the output file, so the output would look something like:
```
class IFace(Interface): ...

<other interfaces>

class IFaceResource(Resource): ...

< other resources>

class IFaceProxy(Proxy[IFace]): ...

<other proxies>

class IFaceGlobal(Global[IFace]): ...

<other globals>

IFace._gen_c()
IFace.resource_class = IFaceResource
IFace.proxy_class = IFaceProxy
IFace.global_class = IFaceGlobal

<other classes>
```
What do you think?